packages/demo-cards/claim.gts

```diff
@@ -53,92 +49,31 @@ class Isolated extends Component<typeof Claim> {
     </CardContainer>
   </template>
 
-  constructor(owner: unknown, args: any) {
-    super(owner, args);
-    this.initialize.perform();
-    if (window.ethereum) {
-      window.ethereum.on('chainChanged', (chainId: string) => {
-        this.connected =
-          parseInt(chainId, 16) == this.args.model.chain?.chainId;
-      });
-    }
-  }
-  private initialize = enqueueTask(async () => {
-    let isSameNetwork = this.isSameNetwork();
-    let isConnected = await this.isMetamaskConnected();
-    this.connected = isConnected && isSameNetwork;
+  // chainId is not explicitly passed to resource
+  // but, the resource is recreated everytime this.chainId changes
+  metamask = getMetamaskResource(this, () => {
+    chainId: this.chainId;
```

Comment:
using .from() to instantiate resource. This resource should regenerate when chainId changes